run_multiple_choice.py

```diff
@@ -239,11 +263,21 @@ def main():
             data_files["train"] = data_args.train_file
         if data_args.validation_file is not None:
             data_files["validation"] = data_args.validation_file
-        extension = data_args.train_file.split(".")[-1]
-        datasets = load_dataset(extension, data_files=data_files)
+
+        if data_args.dataload_file is not None:
+            datasets = load_dataset(data_args.dataload_file, data_files=data_files)
+        else:
+            extension = data_args.train_file.split(".")[-1]
+            datasets = load_dataset(extension, data_files=data_files)
     else:
-        # Downloading and loading the race dataset from the hub.
-        datasets = load_dataset("./datasets/race", "all")
+        # Downloading and loading the dream dataset from the hub.
+        if data_args.dataload_file is not None:
+            if data_args.data_type == 'dream' and data_args.dataload_type != 'plain_text':
+                raise ValueError("When using the 'dream' dataset, the 'dataload_type' parameter should be plain_text.")
```

Comment:
You don't need to do value check here, cuz it has been done in the checking of buider config's name.